Makefile

```diff
@@ -20,14 +22,53 @@ SHELL=/bin/bash
 CWD=$(shell pwd)
 PULL ?=True
 DEV ?=n
-RUN_FLAGS ?=--rm
 CHECK_CHANGED_SCRIPT=$(CWD)/changed_from_main.py
 
+VOLUMES ?=
 ifeq ($(DEV),y)
-	VOLUMES += -v $(CWD):/pace
+	VOLUMES += -v $(ROOT_DIR):/pace
+else
+	VOLUMES += -v $(ROOT_DIR)/test_data:/pace/test_data
 endif
 
+### testing variables
+
+FV3=fv3core
+CONTAINER_ENGINE ?=docker
+RUN_FLAGS ?=--rm
+CONTAINER_CMD?=$(CONTAINER_ENGINE) run $(RUN_FLAGS) $(VOLUMES) $(PACE_IMAGE)
+ifeq ("$(CONTAINER_CMD)","")
+	PACE_PATH?=$(ROOT_DIR)
+else
+	PACE_PATH?=/pace
+endif
+ifeq ("$(CONTAINER_CMD)","")
+	EXPERIMENT_DATA_RUN=$(EXPERIMENT_DATA)
+else
+	EXPERIMENT_DATA_RUN=$(PACE_PATH)/test_data/$(FORTRAN_SERIALIZED_DATA_VERSION)/$(EXPERIMENT)
+endif
+NUM_RANKS ?=6
+MPIRUN_ARGS ?=--oversubscribe
+MPIRUN_CALL ?=mpirun -np $(NUM_RANKS) $(MPIRUN_ARGS)
+TEST_ARGS ?=-v
+TEST_TYPE=$(word 3, $(subst _, ,$(EXPERIMENT)))
+THRESH_ARGS=--threshold_overrides_file=$(PACE_PATH)/fv3core/tests/savepoint/translate/overrides/$(TEST_TYPE).yaml
+
+###
+
+
 build:
+ifneq ($(findstring docker,$(CONTAINER_CMD)),)  # only build if using docker
```

Comment:
I modified this so you can always depend on build and it contains the logic on when to build, instead of every make target having its own copy of the logic on when to build.